src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -528,50 +515,100 @@ private Type EmitExprListNode(AssociativeNode node)
             var arrayTypes = GetTypeStatisticsForArray(eln);
             var ot = GetOverallTypeForArray(arrayTypes);
 
-            var elements = eln.Exprs;
-            
-            EmitOpCode(OpCodes.Ldc_I4, elements.Count);
-            EmitOpCode(OpCodes.Newarr, ot);
-            int elCount = -1;
-            foreach (var el in elements)
+            EmitArray(ot, eln.Exprs, (AssociativeNode el, int idx) => 
             {
-                EmitOpCode(OpCodes.Dup);
-                EmitOpCode(OpCodes.Ldc_I4, ++elCount);
-                var t = DfsTraverse(el);
+                Type t = DfsTraverse(el);
                 if (ot == typeof(object) && t.IsValueType)
                 {
                     EmitOpCode(OpCodes.Box, t);
                 }
-                if (ot == typeof(int))
-                {
-                    EmitOpCode(OpCodes.Stelem_I4);
-                }
-                else if (ot == typeof(long))
-                {
-                    EmitOpCode(OpCodes.Stelem_I8);
-                }
-                else if (ot == typeof(double))
+                if (ot == typeof(double))
                 {
                     if (t == typeof(int) || t == typeof(long))
                     {
                         EmitOpCode(OpCodes.Conv_R8);
```

Comment:
maybe we should add more conversion cases here ?
ex. what if `ot` is long and t is `double`